Lib/test/libregrtest/setup.py

```diff
@@ -109,7 +117,14 @@ def replace_stdout():
     """Set stdout encoder error handler to backslashreplace (as stderr error
     handler) to avoid UnicodeEncodeError when printing a traceback"""
     stdout = sys.stdout
-    sys.stdout = open(stdout.fileno(), 'w',
+    try:
+        fd = stdout.fileno()
+    except io.UnsupportedOperation:
+        # On IDLE, sys.stdout has no file descriptor and is not a TextIOWrapper
+        # object. Leaving sys.stdout unchanged.
+        return
```

Comment:
I'm wondering if we could use `codecs.StreamRecoder` here. But this looks like a more complex issue.

Reply:
I expect that the IDLE shell handles correctly Unicode. replace_stdout() was written for terminals using ASCII/strict for stdout.